### DISCUSSION TITLE
How does the archive assembler create matching distributions?

### DISCUSSION THREAD

**Discussion by helpermethod** [jreleaser/jreleaser#863] [Q&A] [answered]
The documentation says.

`Assemblers will configure a matching distribution by name if it exists and if the assembler exports its artifacts`

When running `assemble`, I can see that a ZIP file is placed under 

    out/jreleaser/assemble/p2e/archive/

Is the ZIP file's name used to configure a distribution? What if I have multiple ZIP files, possibly for each architecture?

Here's what I'm trying to do:

* using JBang to create a native image for each platform
* create an assembly for each native image using JReleaser's archive assembler
* take all the assemblies and release them via JReleaser

This seems to work for linux, but not mac os, although I've created to aptly-named (?) ZIP files.

https://github.com/helpermethod/p2e/blob/main/.github/workflows/release.yml

**Comment by aalmiray** [comment, marked as answer]
Exported assemblers automatically register a distribution whose name matches the name of the assembler. The matching mechanism is independent of the filename or filenames related to the given assembler. 

Archive assemblers automatically register a distribution of type BINARY. In your case you might need the distribution to be NATIVE_IMAGE instead, as that gives you access to NI specific features. 

Given this, I'd recommend to set the archive assembler's `export` property to `false`, and explicitly create a NATIVE_IMAGE distribution (with or without matching name as the assembler is not exported) that defines all target zips as artifacts.

**Comment by helpermethod** [comment]
@aalmiray thanks, it works like a charm ❤️ !
